pattern/equivalence/AlphaEquivalenceTest.java

```diff
@@ -206,48 +234,175 @@ public void test_relation_with_three_roles_equivalent() {
+                parseAnonymousRelationVariable("($y) isa $type"),
+                parseAnonymousRelationVariable("($y) isa $type", "$type type parentship"),
+                parseAnonymousRelationVariable("($y) isa parentship"),
+                parseAnonymousRelationVariable("(parent: $y) isa parentship"),
+                parseAnonymousRelationVariable("(parent: $y) isa $type", "$type type parentship"),
+
+                parseAnonymousRelationVariable("($x, $y)"),
+                parseAnonymousRelationVariable("($x, $y) isa $type"),
+                parseAnonymousRelationVariable("($x, $y) isa parentship"),
+                parseAnonymousRelationVariable("($x, $y) isa $type", "$type type parentship"),
+
+                parseAnonymousRelationVariable("($y, child: $z)"),
+                parseAnonymousRelationVariable("($y, child: $z) isa $type"),
+                parseAnonymousRelationVariable("($y, child: $z) isa parentship"),
+                parseAnonymousRelationVariable("($y, child: $z) isa $type","$type type parentship"),
+
+                parseAnonymousRelationVariable("(parent: $x, child: $y)"),
+                parseAnonymousRelationVariable("(parent: $x, child: $y) isa $type"),
+                parseAnonymousRelationVariable("(parent: $x, child: $y) isa parentship"),
+                parseAnonymousRelationVariable("(parent: $x, child: $y) isa $type","$type type parentship"),
+                parseAnonymousRelationVariable("(parent: $x, child: $y, $z) isa parentship"),
+
+                parseVariables("r0", "$r0 ($y)"),
+                parseVariables("r1", "$r1 ($y) isa $type"),
+                parseVariables("r2", "$r2 (parent: $y) isa parentship"),
+                parseVariables("r3", "$r3 (parent: $y) isa $type","$type type parentship"),
+                parseVariables("r4", "$r4 (parent: $y, child: $z) isa parentship"),
+                parseVariables("r5", "$r5 (parent: $y, child: $z) isa $pship", "$pship type parentship"),
+                parseVariables("r6", "$r6 ($y, $z) isa parentship"),
+                parseVariables("r7", "$r7 (parent: $x, child: $y, $z) isa parentship"),
+                parseVariables("r8", "$r8 ($y, $z, $u) isa parentship")
+        )
+                .map(Variable::asThing)
+                .collect(Collectors.toList());
+        variables.forEach(var -> testAlphaEquivalence(var.asThing(), variables, new HashSet<>()));
+    }
+
+    private void testAlphaEquivalence(ThingVariable varA, ThingVariable varB, boolean isValid){
+        assertTrue(varA.alphaEquals(varA).isValid());
+        assertTrue(varB.alphaEquals(varB).isValid());
+        assertEquals("Variable:\n" + varA + "\n=?\n" + varB, isValid, varA.alphaEquals(varB).isValid());
+        assertEquals("Variable:\n" + varB + "\n=?\n" + varA, isValid, varB.alphaEquals(varA).isValid());
+    }
+
+    private void testAlphaEquivalence(ThingVariable sourceVar, List<ThingVariable> toCheck, Set<Integer> validVars){
```

Comment:
can we give this a better name? is this checking alpha equivalence against itself? (reflexive?)

Reply:
plus better name - `testSymmetricAlphaEquivalence` or something